src/python/pants/backend/python/subsystems/setup.py

```diff
@@ -208,6 +215,21 @@ class PythonSetup(Subsystem):
         ),
         advanced=True,
     )
+    resolves_generate_lockfiles = BoolOption(
+        "--resolves-generate-lockfiles",
+        default=True,
+        help=(
+            "If False, Pants will not attempt to generate lockfiles for `[python].resolves` when "
```

Comment:
I'm not clear on why a user would run generate-lockfiles if using manual resolves. For my manually-generated lockfiles, I'm using "pip freeze" and then tossing that into 3rdparty/python/default_lock.txt (which of course I could rename, but haven't done so yet). I was actually using generate-lockfiles so I could get the metadata, but I now realize I can set invalid_lockfile_behavior=ignore to ignore the error about missing metadata (I found that "warn" still results in an error - if that's incorrect, let me know and I'll make a separate ticket for it). 

So with invalid_lockfile_behavior=ignore, it seems easiest for me to have this workflow:

1. Modify requirements.txt
2. Generate venv using the Pants-recommended script for generating a venv based on requirements.txt (though pip install -r requirements.txt is usually sufficient)
3. pip freeze into 3rdparty/python/default_lock.txt

And of course steps 2 and 3 could be combined into the same script that we could add to our repo and an engineer would run whenever performing step 1. That's fairly painless.

Reply:
> I'm not clear on why a user would run generate-lockfiles if using manual resolves. 

You'd still use it for Pants to generate tool lockfiles for you like `[black].lockfile` if you want to change `[black].version`. Right now, it's really clunky to keep using Pants to generate tool lockfiles for you but tell it to not generate user lockfiles. This PR makes that possible.

> I found that "warn" still results in an error - if that's incorrect, let me know and I'll make a separate ticket for it

Oof, definitely a bug. Please do file!

> And of course steps 2 and 3 could be combined into the same script that we could add to our repo and an engineer would run whenever performing step 1. That's fairly painless.

Yeah, that will be the workflow if doing manual management :)